servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/DefaultHost.java

```diff
@@ -640,4 +648,63 @@ public String toString() {
                     '}';
         }
     }
+
+    private static final class InstrumentedConnectionFactory<Addr, C extends LoadBalancedConnection>
+            extends DelegatingConnectionFactory<Addr, C> {
+
+        private final ConnectTracker connectTracker;
+
+        InstrumentedConnectionFactory(final ConnectionFactory<Addr, C> delegate, ConnectTracker connectTracker) {
+            super(delegate);
+            this.connectTracker = connectTracker;
+        }
+
+        @Override
+        public Single<C> newConnection(Addr addr, @Nullable ContextMap context, @Nullable TransportObserver observer) {
+            return super.newConnection(addr, context, observer).liftSync(delegate -> new ConnectSubscriber(delegate));
+        }
+
+        private class ConnectSubscriber implements SingleSource.Subscriber<C> {
```

Comment:
Instead of using a custom liftSync operator, consider using `defer` to capture `beforeConnectStart` with `beforeFinally` to get the terminal signal.

Reply:
That is much nicer, thank you.